coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java

```diff
@@ -42,11 +43,22 @@ public class CoralSpark {
   private final List<String> baseTables;
   private final List<SparkUDFInfo> sparkUDFInfoList;
   private final String sparkSql;
+  private final HiveMetastoreClient hiveMetastoreClient;
 
+  @Deprecated
   private CoralSpark(List<String> baseTables, List<SparkUDFInfo> sparkUDFInfoList, String sparkSql) {
     this.baseTables = baseTables;
     this.sparkUDFInfoList = sparkUDFInfoList;
     this.sparkSql = sparkSql;
+    this.hiveMetastoreClient = null;
+  }
+
+  private CoralSpark(List<String> baseTables, List<SparkUDFInfo> sparkUDFInfoList, String sparkSql,
+      HiveMetastoreClient mscClient) {
```

Comment:
nit: why is this variable named `mscClient` not `hmsClient`?

Reply:
changed it to `hmsClient` everywhere. I think the idea was to be generic  with `metastoreClient`. However, given how the classes are named and the current supported integration, it's best to call it `hmsClient` for now.